packages/server/lib/hooks/hooks.ts

```diff
@@ -347,9 +347,11 @@ export async function credentialsTest({
             provider,
             providerName: config.provider,
             providerConfigKey: config.unique_key,
-            headers: {
-                'Content-Type': 'application/json'
-            },
+            headers: providerVerification?.skip_automatic_headers
+                ? {}
+                : {
+                      'Content-Type': 'application/json'
+                  },
```

Comment:
I would "just" put content-type to all providers that needs it, the option is too specific to be useful. that's a lot of change but much cleaner